File: 2020/06/06.py
from collections import Counter 

def generator(input):
    return [group.split('\n') for group in input.split('\n\n')]

def part_1(input):
    return solver(input, False)

def part_2(input):
    return solver(input, True)

def solver(input, all):
    return sum([len([c for _, c in Counter(''.join(group)).most_common() if c >= all * len(group)]) for group in input])

File: 2020/06/06.rs
aoc2020::main!();

fn generator(input: &str) -> Vec<Vec<&str>> {
    input
    .split("\n\n")
    .map(|group| group.split('\n').collect_vec())
    .collect_vec()
}

fn part_1(input: Vec<Vec<&str>>) -> usize {
    solver(input, 0)
}

fn part_2(input: Vec<Vec<&str>>) -> usize {
    solver(input, 1)
}

fn solver(groups: Vec<Vec<&str>>, all: usize) -> usize {
    groups
    .iter()
    .map(|group|
        group
        .iter()
        .join("")
        .chars()
        .counts()
        .iter()
        .filter(|(_,n)|
            **n >= all * group.len())
        .count())
    .sum()
}